governance/default.go

```diff
@@ -521,18 +521,60 @@ func (g *Governance) ClearVotes(num uint64) {
 // parseVoteValue parse vote.Value from []uint8 to appropriate type
 func (g *Governance) ParseVoteValue(gVote *GovernanceVote) (*GovernanceVote, error) {
 	var val interface{}
-	k := GovernanceKeyMap[gVote.Key]
+	k, ok := GovernanceKeyMap[gVote.Key]
+	if !ok {
+		logger.Warn("Unknown key was given", "key", k)
+		return nil, ErrUnknownKey
+	}
+
+	// checking type of gVote.Value
+	switch k {
+	case params.AddValidator, params.RemoveValidator:
+		if reflect.TypeOf(gVote.Value) != reflect.TypeOf([]interface{}{}) && reflect.TypeOf(gVote.Value) != reflect.TypeOf([]uint8{}) {
+			return nil, ErrValueTypeMismatch
+		}
 
-	// filter out if vote value is an interface list
-	if reflect.TypeOf(gVote.Value) == reflect.TypeOf([]interface{}{}) {
-		return nil, ErrValueTypeMismatch
+		// if value contains multiple addresses, gVote.Value type should be [][]uint8{}
+		if reflect.TypeOf(gVote.Value) == reflect.TypeOf([]interface{}{}) {
+			for _, item := range gVote.Value.([]interface{}) {
+				if reflect.TypeOf(item) != reflect.TypeOf([]uint8{}) {
+					return nil, ErrValueTypeMismatch
+				}
+				if len(item.([]uint8)) != common.AddressLength {
+					return nil, ErrValueTypeMismatch
+				}
+			}
+			break
+		}
+		// if value contains single addresses, gVote.Value type should be []uint8{}
+		fallthrough
+	default:
+		// Neither governance.addvalidator key nor governance.removevalidator key, gVote.Value type should be []uint8{}
+		if reflect.TypeOf(gVote.Value) != reflect.TypeOf([]uint8{}) {
+			return nil, ErrValueTypeMismatch
+		}
+		if GovernanceItems[k].checkValueType(common.Address{}) && len(gVote.Value.([]uint8)) != common.AddressLength {
+			return nil, ErrValueTypeMismatch
+		}
 	}
 
 	switch k {
 	case params.GovernanceMode, params.MintingAmount, params.MinimumStake, params.Ratio:
 		val = string(gVote.Value.([]uint8))
-	case params.GoverningNode, params.AddValidator, params.RemoveValidator:
+	case params.GoverningNode:
```

Comment:
Hmm.. after updating like this, I'm curious that do we really have two switch statements? we can check the casting failure here to detect type mismatch like:

```go
val, ok = string(gVote.Value.([]uint8))
```

Reply:
Thanks, reflected at https://github.com/klaytn/klaytn/pull/1110/commits/70ad97142c82c7962c5150839e6fb074ae5091f3.

Reply:
@kjhman21 I changed the code to integrate type mismatch checking logic with parsing logic. Please take another look. https://github.com/klaytn/klaytn/pull/1110/commits/e28a478c428d7627141a5c9583d47b3dde921028